### PR TITLE
blockchain/vm: add chainId opcode (eip1344)

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- This PR contains istanbul hardfork item. It is not activated yet since istanbulCompatible block number is set nil.
- This PR is derived from [eip1344](https://github.com/ethereum/go-ethereum/pull/19921).
- ChainId has been supported since Solidity 8.0. If you're using solidity under 8.0, you can get the chainId by calling 0x46 opcode in assembly code.

solidity 8.0 example code
```
pragma solidity >=0.7.0 <0.9.0;

contract ChainIDTest {
    function blockCall() public view returns (uint) {
        return block.chainid;
    }
    function assemblyCall() public view returns (uint256) {
        uint256 id;
        assembly {
            id := chainid()
        }
        return id;
    }
}
```
Result (chainid is 940625)
Screen-Shot-2021-04-26-at-5 31 08-PM

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/vm/opcodes.go [hunk 1]
```diff
@@ -103,6 +103,7 @@ const (
 	NUMBER
 	DIFFICULTY
 	GASLIMIT
+	CHAINID OpCode = 0x46
```
Comment (KimKyungup) [inline blockchain/vm/opcodes.go:106]: This is same with below? @yoomee1313 
```suggestion
	CHAINID OpCode = 0x46
```

**Review by KimKyungup — APPROVED** [review]

File: blockchain/vm/opcodes.go (hunk 1 above)
Comment (yoomee1313) [reply blockchain/vm/opcodes.go:106]: @KimKyungup Yes, both expressions are the same.

Comment (ehnuje) [reply blockchain/vm/opcodes.go:106]: If so, why do you explicitly write `OpCode`? It is also different from geth.

File: blockchain/vm/jump_table.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ type JumpTable [256]operation
 
 func newIstanbulInstructionSet() JumpTable {
 	instructionSet := newConstantinopleInstructionSet()
-	// TODO-IstanbulCompatible: add istanbul hardfork items
+	EnableEIP(1344, &instructionSet)
```
Comment (ehnuje) [inline blockchain/vm/jump_table.go:74]: An error is not handled here.

Comment (aidan-kwon) [reply blockchain/vm/jump_table.go:74]: @yoomee1313 We can call `enable1344` directly as geth do.

**Review by winnie-byun — APPROVED** [review]

File: blockchain/vm/opcodes.go (hunk 1 above)
Comment (yoomee1313) [reply blockchain/vm/opcodes.go:106]: @ehnuje I referenced latest geth code. I should have mentioned it.. https://github.com/ethereum/go-ethereum/blob/d9c9ee5ac9631c434380f73b6e6e50030a04f8ee/core/vm/opcodes.go#L104

File: blockchain/vm/jump_table.go (hunk 2 above)
Comment (yoomee1313) [reply blockchain/vm/jump_table.go]: @ehnuje @aidan-kwon thanks! i edited to call `enable1344`. https://github.com/klaytn/klaytn/pull/952/commits/4bbb9f2c6b504e74357d42d343be196b952abdee.

**Review by KimKyungup — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

**Review by winnie-byun — APPROVED** [review]

**Pull request merged**